cpp/ycm/ClangCompleter/CompletionData.cpp

```diff
@@ -161,12 +169,18 @@ CompletionData::CompletionData( const CXCompletionResult &completion_result ) {
   uint num_chunks = clang_getNumCompletionChunks( completion_string );
   bool saw_left_paren = false;
   bool saw_function_params = false;
+  bool saw_placeholder = false;
 
   for ( uint j = 0; j < num_chunks; ++j ) {
     ExtractDataFromChunk( completion_string,
                           j,
                           saw_left_paren,
-                          saw_function_params );
+                          saw_function_params,
+                          saw_placeholder);
+  }
+  if (original_string_.size()>0 && 
```

Comment:
You need spaces inside IF parens, inside `[]` and spaces around the `>` operator.

Similar for the `erase` call.